s/elements/bindings-editor/views/text-editor-panel.ts

```diff
@@ -2,22 +2,53 @@
 import {html} from "lit"
 import {view} from "@chasemoskal/magical/x/view/view.js"
 
-import {Bindings} from "../../../types.js"
-import {defaultBindingsText} from "../../context/parts/default-bindings.js"
+import {Bindings} from "../../../bindings/types.js"
+import {parseBindings} from "../../../bindings/parse.js"
+import {buildBindings} from "../../../bindings/build.js"
 
 export const TextEditorPanelView = view(use => ({
 		bindings,
+		onClickSave,
 	}: {
 		bindings: Bindings
+		onClickSave: (draft: Bindings) => void
 	}) => {
 
+	const [error, setError] = use.state<undefined | Error>(undefined)
+	const [draft, setDraft] = use.state<undefined | Bindings>(undefined)
+
+	function onTextAreaChange(event: InputEvent) {
+		const target = <HTMLTextAreaElement>event.target
+		try {
+			setDraft(
+				parseBindings(target.value)
+			)
+		}
+		catch (error) {
+			if (error instanceof Error)
+				setError(error)
+		}
+	}
```

Comment:
whenever user changes the textarea input, we parse the bindings to save the draft, and record any errors